app/scripts/controllers/UsernamePromptCtrl.js

```diff
@@ -0,0 +1,22 @@
+(function(){
+
+  function UsernamePromptCtrl($uibModalInstance, $cookieStore) {
+
+
+    this.submit = function() {
+      if(this.text) {
+
+        $cookieStore.put('blocChatCurrentUser', this.text);
```

Comment:
That seems logical based on our last discussion.. but in this case the problem is.. what if you need to access the current user from a different page besides the Username prompt?  It's likely something you need accessible within multiple parts of your application.